src/engine/Operation.cpp

```diff
@@ -96,25 +117,36 @@ ProtoResult Operation::runComputation(const ad_utility::Timer& timer,
   // correct runtimeInfo. The children of the runtime info are already set
   // correctly because the result was computed, so we can pass `nullopt` as
   // the last argument.
-  if (result.isDataEvaluated()) {
+  if (result.isFullyMaterialized()) {
     updateRuntimeInformationOnSuccess(result.idTable().size(),
                                       ad_utility::CacheStatus::computed,
                                       timer.msecs(), std::nullopt);
   } else {
     runtimeInfo().status_ = RuntimeInformation::lazilyMaterialized;
-    result.runOnNewChunkComputed([this, isRoot](
-                                     const IdTable& idTable,
-                                     std::chrono::milliseconds duration) {
-      runtimeInfo().totalTime_ += duration;
-      runtimeInfo().originalOperationTime_ = runtimeInfo().getOperationTime();
-      runtimeInfo().numRows_ += idTable.numRows();
-      runtimeInfo().numCols_ = idTable.numColumns();
-      LOG(DEBUG) << "Computed partial chunk of size " << idTable.numRows()
-                 << " x " << idTable.numColumns() << std::endl;
-      if (isRoot) {
-        signalQueryUpdate();
-      }
-    });
+    result.runOnNewChunkComputed(
+        [this, overlap = 0us, timeSizeUpdate = 0us](
+            const IdTable& idTable,
+            std::chrono::microseconds duration) mutable {
+          overlap += duration;
+          timeSizeUpdate += duration;
+          auto msPrecision =
+              std::chrono::duration_cast<std::chrono::milliseconds>(overlap);
+          updateRuntimeStats(false, idTable.numRows(), idTable.numColumns(),
+                             msPrecision);
+          overlap -= msPrecision;
+          LOG(DEBUG) << "Computed partial chunk of size " << idTable.numRows()
+                     << " x " << idTable.numColumns() << std::endl;
+          if (timeSizeUpdate > 50ms) {
+            timeSizeUpdate = 0us;
+            signalQueryUpdate();
+          }
+        },
+        [this](bool failed) {
+          if (failed) {
```

Comment:
This rather lengthy lambda, which only depends on `this` can be a member function
(i.e. a separate member function that returns the lambda), then you can give it a speaking name and document it without letting the code become too long.